recordset.go

```diff
@@ -197,14 +201,19 @@ func (rcs *Recordset) Results() <-chan *Result {
 // Close all streams from different nodes. A successful close return nil,
 // subsequent calls to the method will return ErrRecordsetClosed.
 func (rcs *Recordset) Close() error {
+	// check and close cancelled channel
+	// this will broadcast to all commands listening to the channel
+	select {
+	case <-rcs.cancelled:
+	default:
+		close(rcs.cancelled)
+	}
+
```

Comment:
The issue is that in signalEnd() when the goroutines count reached 0 then rcs.active.Set gets called. This causes the close of the cancelled channel to get skipped when Close() is called (because it is after the rcs.active.CompareAndToggle(true) call).

So essentially in the current code cancelled channel gets closed once only if signalEnd() hasn't already finished. If signalEnd() finishes before Close() then cancelled never gets closed and goroutines start leaking.

The block I added ensures cancelled get closed (only once) but is not blocked by the state of active.